processing/src/main/java/io/druid/segment/FloatDimensionIndexer.java

```diff
@@ -97,13 +102,20 @@ public float getFloat()
       {
         final Object[] dims = currEntry.get().getDims();
 
-        if (dimIndex >= dims.length) {
+        if (dimIndex >= dims.length || dims[dimIndex] == null) {
           return 0.0f;
         }
 
         return (Float) dims[dimIndex];
       }
 
+      @Override
```

Comment:
Align this class with Double version. E. g. in Double version methods go in different order, and the logic is different.

Reply:
done